lib/sycamore/sycamore/evaluation/pipeline.py

```diff
@@ -36,46 +41,60 @@ def __init__(
         else:
             self._query_executor = query_executor
         self._os_config = os_config
+        self._subtask_docs = subtask_docs
+        self._embedder = embedder
 
     def _add_filter(self, query_body: dict, filters: dict[str, str]):
-        hybrid_query_match = query_body["query"]["hybrid"]["queries"][0]
-        hybrid_query_match = {
-            "bool": {
-                "must": [hybrid_query_match],
-                "filter": [{"match_phrase": {f"{k}.keyword": filters[k]}} for k in filters],
-            }
-        }
-        query_body["query"]["hybrid"]["queries"][0] = hybrid_query_match
-        hybrid_query_neural = query_body["query"]["hybrid"]["queries"][1]
-        hybrid_query_neural["neural"]["embedding"]["filter"] = {
-            "bool": {"must": [{"match_phrase": {k: filters[k]}} for k in filters]}
-        }
-        query_body["query"]["hybrid"]["queries"][1] = hybrid_query_neural
+        hybrid_query_match = query_body["query"]["knn"]["embedding"]["filter"]["bool"]["must"]
+        for key, val in filters.items():
+            hybrid_query_match.append({"match_phrase": {key: val}})
+        query_body["query"]["knn"]["embedding"]["filter"]["bool"]["must"] = hybrid_query_match
         return query_body
 
     def _build_opensearch_query(self, doc: Document) -> Document:
         assert doc.type == "EvaluationDataPoint"
         query = OpenSearchQuery(doc)
         query["index"] = self._index
+
+        if self._subtask_docs and doc.properties["subtasks_reqd"]:
+            filtered_docs = [
+                st_doc for st_doc in self._subtask_docs if st_doc.parent_id == doc["raw"]["financebench_id"]
```

Comment:
maybe make "financebench_id" parametrizable or more general?

Reply:
hmm - `self._subtask_docs` is a DocSet, right? I don't think you can just iterate over a DocSet like this. Perhaps you want `filtered_docs = self._subtask_docs.filter(lambda st_doc: st_doc.parent_id == doc['raw']['financebench_id']`?

Reply:
fixed the financebench_id

`self._subtask_docs` is a list of Documents (the type returned by the SubtaskExecutor `execute` needs to be corrected). filtering a DocSet would require a `take_all` for every question, so it seemed cleaner to have just one `take_all` and execute all of the subtasks before they are passed into the EvalPipeline.